.github/workflows/test-nightly.yml

```
@@ -80,4 +78,19 @@ jobs:
           CELESTIA_TAG: ${{ steps.tag.outputs.value }}
         run: make test-docker-e2e test=TestAllUpgrades
 
-      # TODO: add slack notification on failure
+  notify-slack-on-failure:
+    name: Notify Slack on failure
+    runs-on: ubuntu-latest
+    needs: [test-state-sync-mocha, test-all-upgrades]
+    if: ${{ always() && (needs.test-state-sync-mocha.result == 'failure' || needs.test-all-upgrades.result == 'failure') }}
+    steps:
+      - name: Slack Notification
+        uses: slackapi/slack-github-action@v1.27.0
+        with:
+          payload: |
+            {
+              "text": "🚨 Nightly E2E Tests Failed: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```

[nit] can we please avoid emojis in this log?

```suggestion
              "text": "test-nightly tests failed: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```